public/app/percona/dbaas/components/DBCluster/EditDBClusterModal/DBClusterAdvancedOptions/DBClusterAdvancedOptions.tsx

```diff
@@ -79,7 +81,9 @@ export const DBClusterAdvancedOptions: FC<DBClusterAdvancedOptionsProps> = ({ se
         setLoadingAllocatedResources(false);
       }
 
-      allocatedTimer = setTimeout(() => getAllocatedResources(false), RECHECK_INTERVAL);
+      if (!unmounted.current) {
+        allocatedTimer = setTimeout(() => getAllocatedResources(false), RECHECK_INTERVAL);
```

Comment:
Is this to avoid state being set after the component is unmounted and a request resolves? If so, using the cancel token and `isApiCancelError()` without `finally` should take care of this without having to manually check if the component has unmounted (see `BackupInventory`, for example).

Reply:
No, is to prevent to do more requests after the component unmounts. I'm doing a new request when the previous finishes, if the component unmounts when there is a pending request it will keep doing requests "forever". Cancelling the request works only if I move the code that triggers a new request to the `try` block, but I want to trigger a new request independent of the result (if it succeeds or not) and that wouldn't work in that case. To make it work with the cancelling I would have to call a new request both inside the `try` block and inside the `catch`, but in the latter would have to check if it was cancelled or not. Seems like it complicates a little more than the current solution :/

Reply:
👍